File: 201902646/10866.cpp
#include <iostream>
#include <deque>
#include <string>

int main() {

	std::deque<int> deq;
	std::string cmd;

	int N;
	std::cin >> N;
	int X;

	for (int i = 0; i < N; ++i) {
		std::cin >> cmd;
		if (cmd == "push_front") {
			std::cin >> X;
			deq.push_front(X);
		}
		else if (cmd == "push_back") {
			std::cin >> X;
			deq.push_back(X);
		}
		else if (cmd == "pop_front") {
			if (deq.empty()) {
				std::cout << -1 << std::endl;
			}
			else {
				std::cout << deq.front() << std::endl;
				deq.pop_front();
			}
		}
		else if (cmd == "pop_back") {
			if (deq.empty()) {
				std::cout << -1 << std::endl;
			}
			else {
				std::cout << deq.back() << std::endl;
				deq.pop_back();
			}
		}
		else if (cmd == "size") {
			std::cout << deq.size() << std::endl;
		}
		else if (cmd == "empty") {
			if (deq.empty()) {
				std::cout << 1 << std::endl;
			}
			else {
				std::cout << 0 << std::endl;
			}
		}
		else if (cmd == "front") {
			if (deq.empty()) {
				std::cout << -1 << std::endl;
			}
			else {
				std::cout << deq.front() << std::endl;
			}
		}
		else if (cmd == "back") {
			if (deq.empty()) {
				std::cout << -1 << std::endl;
			}
			else {
				std::cout << deq.back() << std::endl;
			}
		}
	}

	return 0;
}

File: 201902693/10866.py
import sys
input = sys.stdin.readline
n = int(input())
deque = []
for i in range(n) :
    cmd = list(input().split())
    if cmd[0] == 'push_front' :
        deque = [int(cmd[1])] + deque
    if cmd[0] == 'push_back' :
        deque.append(int(cmd[1]))
    if cmd[0] == 'pop_front' :
        if len(deque) == 0 :
            print(-1)
        else :
            print(deque[0])
            del deque[0]
    if cmd[0] == 'pop_back' :
        if len(deque) == 0 :
            print(-1)
        else :
            print(deque[-1])
            del deque[-1]
    if cmd[0] == 'size' :
        print(len(deque))
    if cmd[0] == 'empty' :
        if len(deque) == 0 :
            print(1)
        else :
            print(0)
    if cmd[0] == 'front' :
        if len(deque) == 0 :
            print(-1)
        else :
            print(deque[0])
    if cmd[0] == 'back' :
        if len(deque) == 0 :
            print(-1)
        else :
            print(deque[-1])    